### PR TITLE
Support including job files from a directory

### LINKED ISSUE #2: Feature: include extra jobs
Tarsnapper is extremely useful, thanks!

I would find it very convenient if there were a way to specify job in separate files, something like:

    # Global values, valid for all jobs unless overridden:
    deltas: 1d 7d 30d
    target: /localmachine/$name-$date
    extra_jobs: /usr/local/etc/tarsnapper/extra/*
   
    jobs:
      # some jobs

and then have one or more jobs per file in the given directory.

The rationale for this is, I use Chef and I have created my own tarsnap cookbooks.
This feature would make it easy for other recipes (e.g. a database) to configure their jobs, which would get picked up automatically.

I would send a pull request but my knowledge of Python is next to 0 :)

### PR DESCRIPTION
**Pull request by caspark**
Fixes #2.

Also:

* Enforce globally unique job names instead of using last-one-wins
  semantics
* Update documentation with some more explanations and comments
* Add missing code block formatting in documentation
* Use PyYaml's safe_load() instead of load()

### CODE REVIEW

**Comment by caspark** [comment]
I'm not a huge fan of the hyphen in `include-jobs`, but I'm following the precedent set by `delta-names`.

This PR also supports the `include-jobs: /var/lib/vz/private/*/etc/backup.conf` approach that [you mentioned you were after](https://github.com/miracle2k/tarsnapper/issues/2#issuecomment-1015233) years ago, as tested with `include-jobs: /home/caspar/src/tarsnapper/more-jobs/*/blah/*` and a file structure of

```
$ find more-jobs/
more-jobs/
more-jobs/a
more-jobs/a/blah
more-jobs/a/blah/001-important-jobs.yml
more-jobs/b
more-jobs/b/blah
more-jobs/b/blah/020-other-jobs.yml
more-jobs/b/blah/www.yml
```

(all the jobs in yml files get picked up)

This PR is lacking in tests; I went for the approach which minimizes invasive changes to the config module, which leads to accessing the filesystem in `load_config`, which is tricky to test.

File: src/tarsnapper/config.py [hunk 1]
```diff
@@ -112,7 +128,7 @@ def load_config(text):
     """Load the config file and return a dict of jobs, with the local
     and global configurations merged.
     """
-    config = yaml.load(text)
+    config = yaml.safe_load(text)
```
Comment (caspark) [inline src/tarsnapper/config.py]: I think `load` is sufficient here, so I made an unrelated fix.

File: README.rst [hunk 2]
```diff
@@ -138,7 +165,7 @@ The most recent backup is always kept.
 
 As an example, here is a list of backups from a Desktop computer that has
 often been running non-stop for days, but also has on occasion been turned
-off for weeks at a time, using the deltas ``1d 7d 30d 360d 18000d``:
+off for weeks at a time, using the deltas ``1d 7d 30d 360d 18000d``::
```
Comment (caspark) [inline README.rst]: Unrelated formatting fix.

**Comment by miracle2k** [comment]
Excellent, looks great. Thanks!

**Comment by jeffwidman** [comment]
Much thanks @caspark - this will make a few of my Ansible playbooks much simpler.

**Comment by caspark** [comment]
@jeffwidman you're welcome; I actually implemented it for use with [pmbauer/ansible-tarsnap](https://github.com/pmbauer/ansible-tarsnap) myself! (so am also waiting for a Pypi release ;) )

**Comment by miracle2k** [comment]
I pushed a new release to PyPi.

**Comment by jeffwidman** [comment]
Thanks!
